src/display.c

```diff
@@ -360,22 +360,15 @@ PyImaging_GrabScreenWin32(PyObject *self, PyObject *args) {
     SetThreadDpiAwarenessContext_function = (Func_SetThreadDpiAwarenessContext
     )GetProcAddress(user32, "SetThreadDpiAwarenessContext");
     if (SetThreadDpiAwarenessContext_function != NULL) {
-        if (screens == -1) {
-            GetWindowDpiAwarenessContext_function = (Func_GetWindowDpiAwarenessContext
-            )GetProcAddress(user32, "GetWindowDpiAwarenessContext");
-            DPI_AWARENESS_CONTEXT dpiAwarenessContext =
-                GetWindowDpiAwarenessContext_function(wnd);
-            if (GetWindowDpiAwarenessContext_function != NULL &&
-                dpiAwarenessContext != NULL) {
-                dpiAwareness =
-                    SetThreadDpiAwarenessContext_function(dpiAwarenessContext);
-            } else {
-                dpiAwareness = SetThreadDpiAwarenessContext_function((HANDLE)-3);
-            }
-        } else {
-            // DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE = ((DPI_CONTEXT_HANDLE)-3)
-            dpiAwareness = SetThreadDpiAwarenessContext_function((HANDLE)-3);
+        GetWindowDpiAwarenessContext_function = (Func_GetWindowDpiAwarenessContext
+        )GetProcAddress(user32, "GetWindowDpiAwarenessContext");
+        if (screens == -1 && GetWindowDpiAwarenessContext_function != NULL) {
+            dpiAwareness = GetWindowDpiAwarenessContext_function(wnd);
```

Comment:
```suggestion
        if (screens == -1) {
            GetWindowDpiAwarenessContext_function = (Func_GetWindowDpiAwarenessContext
            )GetProcAddress(user32, "GetWindowDpiAwarenessContext");
            if (GetWindowDpiAwarenessContext_function != NULL) {
                dpiAwareness = GetWindowDpiAwarenessContext_function(wnd);
            }
```
So your feeling is that your current version is simpler than this suggestion? I think your version is also probably fine.

Reply:
I generally find fewer nested if statements to be easier to read.